### ISSUE TITLE
"AttributeError: project" when call registry.list_saved_datasets

### ISSUE THREAD

**Issue by zeroprop** [feast-dev/feast#3881]
I'm trying print all datasets in my project.
`print(store.registry.list_saved_datasets('PROJECT_NAME'))`
## Expected Behavior 
List of all datasets in project
## Current Behavior
```
Traceback (most recent call last):
  File "C:\Users\user\PycharmProjects\feast-service\test.py", line 64, in <module>
    enriched_df = prepare_dataset(store, df)
  File "C:\Users\user\PycharmProjects\feast-service\test.py", line 43, in prepare_dataset
    print(store.registry.list_saved_datasets('PROJECT_NAME'))
  File "C:\Users\user\PycharmProjects\feast-service\venv\lib\site-packages\feast\infra\registry\registry.py", line 696, in list_saved_datasets
    return proto_registry_utils.list_saved_datasets(registry_proto, project)
  File "C:\Users\user\PycharmProjects\feast-service\venv\lib\site-packages\feast\infra\registry\proto_registry_utils.py", line 217, in list_saved_datasets
    if saved_dataset.project == project:
AttributeError: project
```
## Steps to reproduce

1. Create Feature Store
2. Call registry.list_saved_datasets method

### Specifications

- Version:
   feast==0.31.1
- Platform:
   Windows

## Possible Solution
I think in `list_saved_datasets()` we should use `saved_dataset.spec.project` instead of `saved_dataset.project`. As in `get_saved_dataset()` method
```
def get_saved_dataset(
    registry_proto: RegistryProto, name: str, project: str
) -> SavedDataset:
    for saved_dataset in registry_proto.saved_datasets:
        if saved_dataset.spec.name == name and saved_dataset.spec.project == project:
            return SavedDataset.from_proto(saved_dataset)
    raise SavedDatasetNotFound(name, project=project)
```

**Comment by tokoko** [comment]
This is already fixed in the latest version after #3717

**Comment by jeremyary** [comment]
ack on fix w/3717. Thanks!
